core/src/main/scala/org/broadinstitute/dsp/workbench/welder/BackgroundTask.scala

```diff
@@ -28,23 +35,44 @@ object BackgroundTask {
       _ <- logger.debug(s"updated metadata cache ${updatedMap}")
     } yield ()).handleErrorWith(t => logger.error(t)("fail to update metadata cache"))
 
-    (Stream.sleep[IO](interval) ++ Stream.eval(task)).repeat
+    (Stream.sleep[IO](config.cleanUpLockInterval) ++ Stream.eval(task)).repeat
   }
 
   def flushBothCache(
-      interval: FiniteDuration,
       storageLinksPath: Path,
       metadataCachePath: Path,
-      storageLinksCache: StorageLinksCache,
-      metadataCache: MetadataCache,
       blockingEc: ExecutionContext
-  )(implicit cs: ContextShift[IO], logger: Logger[IO], timer: Timer[IO]): Stream[IO, Unit] = {
+  ): Stream[IO, Unit] = {
     val flushStorageLinks = flushCache(storageLinksPath, blockingEc, storageLinksCache).handleErrorWith { t =>
       Stream.eval(logger.info(t)("failed to flush storagelinks cache to disk"))
     }
     val flushMetadataCache = flushCache(metadataCachePath, blockingEc, metadataCache).handleErrorWith { t =>
       Stream.eval(logger.info(t)("failed to flush metadata cache to disk"))
     }
-    (Stream.sleep[IO](interval) ++ flushStorageLinks ++ flushMetadataCache).repeat
+    (Stream.sleep[IO](config.flushCacheInterval) ++ flushStorageLinks ++ flushMetadataCache).repeat
+  }
+
+  val syncCloudStorageDirectory: Stream[IO, Unit] = {
+    val res = for {
+      storageLinks <- storageLinksCache.get
+      traceId <- IO(TraceId(UUID.randomUUID()))
+      _ <- storageLinks.values.toList.traverse { storageLink =>
+        logger.info(s"syncing file from ${storageLink.cloudStorageDirectory}") >>
+        (googleStorageAlg
+          .localizeCloudDirectory(storageLink.localBaseDirectory.path, storageLink.cloudStorageDirectory, config.workingDirectory, traceId)
```

Comment:
Do we need to honor the storage link `pattern` here?

Reply:
ah yes! thanks for catching this

Reply:
resolved